common/core/nesteddisplay.cpp

```diff
@@ -195,97 +195,97 @@ bool NestedDisplay::Present(std::vector<HwcLayer *> &source_layers,
-    uint32_t surf_index = 0;
-    HwcLayer *layer = source_layers.at(layer_index);
-    if (!layer->IsVisible())
-      continue;
-
-    const HwcRect<int> &display_frame = layer->GetDisplayFrame();
-    sf_handle = layer->GetNativeHandle();
-    auto search = mHyperDmaExportedBuffers.find(sf_handle);
-    if (search == mHyperDmaExportedBuffers.end()) {
-      std::shared_ptr<OverlayBuffer> buffer(NULL);
-      buffer = OverlayBuffer::CreateOverlayBuffer();
-      buffer->InitializeFromNativeHandle(sf_handle, resource_manager_.get());
-
-      if (mHyperDmaBuf_Fd > 0 && buffer->GetPrimeFD() > 0) {
-        struct ioctl_hyper_dmabuf_export_remote msg;
-        memset(&msg, 0, sizeof(ioctl_hyper_dmabuf_export_remote));
-
-        /* TODO: add more flexibility here, instead of hardcoded domain 0*/
-        msg.remote_domain = 0;
-        msg.dmabuf_fd = buffer->GetPrimeFD();
-
-        ret = ioctl(mHyperDmaBuf_Fd, IOCTL_HYPER_DMABUF_EXPORT_REMOTE, &msg);
-        if (ret) {
-          ETRACE("Hyper DmaBuf: Exporting hyper_dmabuf failed with error %d\n",
-                 ret);
-          return false;
-        } else {
-          ETRACE("Hyper DmaBuf: Exporting hyper_dmabuf Done! 0x%x\n",
-                 msg.hid.id);
-          mHyperDmaExportedBuffers[sf_handle].surf_index = surf_index++;
-          mHyperDmaExportedBuffers[sf_handle].width = buffer->GetWidth();
-          mHyperDmaExportedBuffers[sf_handle].height = buffer->GetHeight();
-          mHyperDmaExportedBuffers[sf_handle].format = buffer->GetFormat();
-          pitches = buffer->GetPitches();
-          offsets = buffer->GetOffsets();
-          mHyperDmaExportedBuffers[sf_handle].pitch[0] = pitches[0];
-          mHyperDmaExportedBuffers[sf_handle].pitch[1] = pitches[1];
-          mHyperDmaExportedBuffers[sf_handle].pitch[2] = pitches[2];
-          mHyperDmaExportedBuffers[sf_handle].offset[0] = offsets[0];
-          mHyperDmaExportedBuffers[sf_handle].offset[1] = offsets[1];
-          mHyperDmaExportedBuffers[sf_handle].offset[2] = offsets[2];
-          mHyperDmaExportedBuffers[sf_handle].tile_format =
-              buffer->GetTilingMode();
-          mHyperDmaExportedBuffers[sf_handle].rotation = 0;
-          mHyperDmaExportedBuffers[sf_handle].status = 0;
-          mHyperDmaExportedBuffers[sf_handle].counter = 0;
-          mHyperDmaExportedBuffers[sf_handle].hyper_dmabuf_id = msg.hid;
-          mHyperDmaExportedBuffers[sf_handle].surface_id = (uint64_t)sf_handle;
-          strncpy(mHyperDmaExportedBuffers[sf_handle].surface_name, "Cluster",
-                  SURFACE_NAME_LENGTH);
-          mHyperDmaExportedBuffers[sf_handle].bbox[0] = display_frame.left;
-          mHyperDmaExportedBuffers[sf_handle].bbox[1] = display_frame.top;
-          mHyperDmaExportedBuffers[sf_handle].bbox[2] = buffer->GetWidth();
-          mHyperDmaExportedBuffers[sf_handle].bbox[3] = buffer->GetHeight();
+  if (mconnected) {
```

Comment:
just do an early return here ? indentation needs to be fixed.

Reply:
yes, if no sharing request from SOS side, hwc will mark the nesteddisplay's layers to device type, then do nothing . While got sharing request from SOS side, hwc will leverage surfaceflinger do composition, then share the FB to SOS.

Reply:
I mean u could just have if (!mconnected)
                                          return false;